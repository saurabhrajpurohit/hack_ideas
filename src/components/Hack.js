import PropTypes from 'prop-types';
import { Card, Typography, Stack, Chip } from '@mui/material';
import { Recommend } from '@mui/icons-material';

Hack.propTypes = {
    hackathon: PropTypes.object,
    handleClick: PropTypes.func,
    userId: PropTypes.number
};

export default function Hack({ hackathon, handleClick, userId }) {
    const { Title, Description, EventDate, Tags, Votes } = hackathon;
    const isLiked = Votes.includes(userId);
    return (
        <Card elevation={4}>
            <Stack spacing={1} sx={{ p: 3 }}>
                <Stack direction="row" alignItems="center" justifyContent="space-between">
                    <Typography variant="h6">
                        {Title}
                    </Typography>
                    <Chip
                        data-testid="upvote-element"
                        label={Votes.length}
                        size="small"
                        onClick={() => handleClick(!isLiked)}
                        icon={<Recommend />}
                        variant="outlined"
                        color={isLiked ? 'primary' : 'default'}
                    />
                </Stack>
                <Stack direction="row" alignItems="center" justifyContent="space-between">
                    <Typography variant="caption">
                        Event On: &nbsp;
                        <Typography
                            component="span"
                            variant="overline"
                            sx={{
                                color: 'text.primary',
                            }}
                        >
                            {new Date(EventDate.toDate()).toDateString()}
                        </Typography>
                    </Typography>
                </Stack>
                <Stack direction="row" spacing={1}>
                    {Tags.map(tag => (<Chip key={tag} size="small" color="success" label={tag} variant="outlined" />))}
                </Stack>
                <Typography variant="subtitle2" gutterBottom>
                    {Description}
                </Typography>
            </Stack>
        </Card >
    );
}
