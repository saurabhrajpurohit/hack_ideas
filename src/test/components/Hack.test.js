import { fireEvent, render, screen } from '@testing-library/react';
import Hack from '../../components/Hack';

const hackathon = {
    id: "test_id",
    Votes: [1001],
    Title: "Test Title",
    Tags: ["Front-End"],
    EventDate: {toDate: () => 1646006400000},
    Description: "Test Description"
}

test('click event is to be called', () => {
    const handleClick = jest.fn()
    render(<Hack hackathon={hackathon} handleClick={handleClick} />);
    const element = screen.getByTestId('upvote-element');
    fireEvent.click(element);
    expect(handleClick).toHaveBeenCalledTimes(1)
});

test('ui to be rendered correctly', () => {
    const handleClick = jest.fn()
    render(<Hack hackathon={hackathon} handleClick={handleClick} />);
    const element = screen.getByTestId('test_id');
    expect(element).toMatchSnapshot();
});
