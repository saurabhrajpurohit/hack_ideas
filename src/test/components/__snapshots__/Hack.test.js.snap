// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui to be rendered correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root"
  data-testid="test_id"
>
  <div
    class="css-1hridrd-MuiStack-root"
  >
    <div
      class="css-1jv2z0n-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Test Title
      </h6>
      <div
        class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1yd5aew-MuiButtonBase-root-MuiChip-root"
        data-testid="upvote-element"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
          data-testid="RecommendIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm6 9.8a.9.9 0 0 1-.1.5l-2.1 4.9a1.34 1.34 0 0 1-1.3.8H9a2 2 0 0 1-2-2v-5a1.28 1.28 0 0 1 .4-1L12 5l.69.69a1.08 1.08 0 0 1 .3.7v.2L12.41 10H17a1 1 0 0 1 1 1z"
          />
        </svg>
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          1
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="css-1jv2z0n-MuiStack-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Event On: Â 
        <span
          class="MuiTypography-root MuiTypography-overline css-1gj9p0o-MuiTypography-root"
        >
          Mon Feb 28 2022
        </span>
      </span>
    </div>
    <div
      class="css-w4z10b-MuiStack-root"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-outlinedSuccess css-1smg64b-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          Front-End
        </span>
      </div>
    </div>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
    >
      Test Description
    </h6>
  </div>
</div>
`;
